sim/monk/mistweaver/mana_tea.go

```diff
@@ -0,0 +1,102 @@
+package mistweaver
+
+import (
+	"math"
+	"time"
+
+	"github.com/wowsims/mop/sim/core"
+	"github.com/wowsims/mop/sim/core/stats"
+	"github.com/wowsims/mop/sim/monk"
+)
+
+func (mw *MistweaverMonk) registerManaTea() {
+
+	buffActionID := core.ActionID{SpellID: 115294}
+	stackActionID := core.ActionID{SpellID: 123766}
+	manaMetrics := mw.NewManaMetrics(buffActionID)
+	manaPerTick := 0.0
+	//numerOFTicks := 6
+
+	mw.Monk.RegisterOnChiSpent(func(sim *core.Simulation, chiSpent int32) {
+		accumulatedChi := mw.outstandingChi + chiSpent
+
+		for accumulatedChi >= 4 {
+
+			mw.AddBrewStacks(sim, 1)
+			accumulatedChi -= 4
+		}
+
+		mw.outstandingChi = accumulatedChi
+
+	})
+
+	mw.ManaTeaStackAura = mw.RegisterAura(core.Aura{
+		Label:     "Mana Tea Stacks" + mw.Label,
+		ActionID:  stackActionID,
+		Duration:  time.Hour,
+		MaxStacks: 10,
+	})
+
+	mw.Monk.RegisterOnNewBrewStacks(func(sim *core.Simulation, stacksToAdd int32) {
+		mw.ManaTeaStackAura.Activate(sim)
+
+		procChance := mw.GetStat(stats.SpellCritPercent)
+
+		if sim.Proc(math.Mod(procChance, 1), "Mana Tea") {
+			stacksToAdd += 1
+		}
```

Comment:
This doesn't seem correct, it would set the proc chance to 99% if your current Spell Crit chance is 1.99% for example.